agent-control/src/cli/on_host/systemd_gen.rs

```diff
@@ -0,0 +1,162 @@
+
+/// Generates the Agent Control configuration for host environments.
+#[derive(Debug, clap::Parser)]
+pub struct Args {
+    /// Sets which host monitoring source to be used.
+    #[arg(long, required = true)]
+    newrelic_license_key: String,
+
+    /// New Relic region
+    #[arg(long, value_parser = region_parser(), required = true)]
+    region: Region,
+}
+
+/// Generates the entries required by agent-control in newrelic-agent-control.conf.
+pub fn generate_systemd_config(args: Args) -> Result<(), CliError> {
+    info!("Adding required values to newrelic-agent-control.conf ");
+
+    update_newrelic_license_key(CONFIG_PATH, &args.newrelic_license_key)?;
+
+    update_otel_exporter_endpoint(CONFIG_PATH, args.region)?;
+
+    info!("Host monitoring values generated successfully");
+    Ok(())
+}
+
+fn update_newrelic_license_key(config_path: &str, new_license_key: &str) -> Result<(), CliError> {
+    let content = std::fs::read_to_string(config_path)
+        .map_err(|err| CliError::Command(format!("error reading agent control .conf file: {err}")))?
+        .lines()
+        .filter(|line| !line.starts_with(NEW_RELIC_LICENSE_KEY))
+        .collect::<Vec<_>>()
+        .join("\n");
+
+    let mut file = OpenOptions::new()
+        .write(true)
+        .truncate(true)
+        .open(config_path)
+        .map_err(|err| {
+            CliError::Command(format!("error opening agent control .conf file: {err}"))
+        })?;
+
+    let new_content = format!(
+        "{}\n{}=\"{}\"\n",
+        content, NEW_RELIC_LICENSE_KEY, new_license_key
+    );
+    file.write_all(new_content.as_bytes()).map_err(|err| {
+        CliError::Command(format!("error updating agent control .conf file: {err}"))
+    })?;
+
+    Ok(())
+}
+
+fn update_otel_exporter_endpoint(config_path: &str, region: Region) -> Result<(), CliError> {
+    let content = std::fs::read_to_string(config_path)
+        .map_err(|err| CliError::Command(format!("error reading agent control .conf file: {err}")))?
+        .lines()
+        .filter(|line| !line.starts_with(OTEL_EXPORTER_OTLP_ENDPOINT))
+        .collect::<Vec<_>>()
+        .join("\n");
+
+    let new_content = format!(
+        "{}\n{}=https://{}:4317/\n",
+        content,
+        OTEL_EXPORTER_OTLP_ENDPOINT,
+        get_otlp_endpoint(region)
+    );
+
+    let mut file = OpenOptions::new()
+        .write(true)
+        .truncate(true)
+        .open(config_path)
+        .map_err(|err| {
+            CliError::Command(format!("error opening agent control .conf file: {err}"))
+        })?;
+
+    file.write_all(new_content.as_bytes()).map_err(|err| {
+        CliError::Command(format!("error updating agent control .conf file: {err}"))
+    })?;
+
+    Ok(())
+}
+
+fn get_otlp_endpoint(region: Region) -> String {
+    let url = match region {
+        Region::STAGING => OTLP_URL_STAGING,
+        Region::EU => OTLP_URL_EU,
+        Region::US => OTLP_URL_US,
+    };
+    url.to_string()
+}
```

Comment:
This is what I was talking about moving to `Region`, I'd say it would easy unit tests.